monkey/infection_monkey/island_api_client/http_island_api_client.py

```diff
@@ -76,3 +78,14 @@ def get_pba_file(self, filename: str):
         response.raise_for_status()
 
         return response.content
+
+    @handle_island_errors
+    def send_events(self, events: Sequence[JSONSerializable]):
```

Comment:
The `IIslandAPIClient` should be responsible for serializing the data. Have it take `Sequence[AbstractAgentEvent]` and move serialization here.